File: scripts/parse_pdf.py
import os
import sys

try:
    from pypdf import PdfReader
except ImportError:
    os.system("pip install pypdf")
    from pypdf import PdfReader

def extract_pdf():
    pdf_path = os.path.join("doc", "DeD_5e_manuale-del-giocatore.pdf")
    if not os.path.exists(pdf_path):
        print(f"File non trovato: {pdf_path}")
        return

    print(f"Apertura di {pdf_path}...")
    reader = PdfReader(pdf_path)
    print(f"Pagine totali: {len(reader.pages)}")
    
    # Extract only from page 18 to 43 (0-indexed so 17 to 42)
    start_page = max(0, 18 - 1)
    end_page = min(len(reader.pages), 43)
    
    text = ""
    for i in range(start_page, end_page):
        page = reader.pages[i]
        text += f"\n--- PAGINA {i + 1} ---\n"
        text += page.extract_text()
        
    with open("tmp-races-classes.txt", "w", encoding="utf-8") as f:
        f.write(text)
        
    print(f"Estratte {end_page - start_page} pagine. Testo salvato in tmp-races-classes.txt")

if __name__ == "__main__":
    extract_pdf()

File: scripts/parse-pdf.js
import fs from "fs";
import _pdf from "pdf-parse";
const pdf = _pdf.default || _pdf;
import path from "path";

async function parsePdf() {
  console.log("Leggendo il PDF...");
  const dataBuffer = fs.readFileSync(path.join(process.cwd(), "doc", "DeD_5e_manuale-del-giocatore.pdf"));

  try {
    const data = await pdf(dataBuffer);
    
    // Quick workaround to read the text and grab pages 18-43 roughly
    console.log(`PDF letto con successo. Caratteri totali: ${data.text.length}`);
    
    // Save the whole text roughly to a tmp file so we can inspect it and write a regex for it
    fs.writeFileSync("tmp-pdf-all.txt", data.text);
    console.log("Testo completo salvato in tmp-pdf-all.txt");
  } catch (error) {
    console.error("Errore durante il parsing del PDF:", error);
  }
}

parsePdf();
